src/mssql_server/mod.rs

```diff
@@ -0,0 +1,162 @@
+use std::collections::HashMap;
+
+use testcontainers::{core::WaitFor, Image};
+
+/// [Microsoft SQL Server](https://www.microsoft.com/en-us/sql-server) module
+/// for [testcontainers](https://crates.io/crates/testcontainers).
+///
+/// This module is based on the
+/// [official Microsoft SQL Server for Linux Docker image](https://hub.docker.com/_/microsoft-mssql-server).
+/// Only amd64 images are available for SQL Server. If you use Apple silicon machines,
+/// you need to configure Rosetta emulation.
+///
+/// * [Change Docker Desktop settings on Mac | Docker Docs](https://docs.docker.com/desktop/settings/mac/#general)
+///
+/// # Example
+///
+/// ```
+/// use testcontainers::clients;
+/// use testcontainers_modules::mssql_server;
+///
+/// let docker = clients::Cli::default();
+/// let mssql_server = docker.run(mssql_server::MssqlServer::default());
+/// let ado_connection_string = format!(
+///    "Server=tcp:127.0.0.1,{};Database=test;User Id=sa;Password=yourStrong(!)Password;TrustServerCertificate=True;",
+///    mssql_server.get_host_port_ipv4(1433)
+/// );
+/// ```
+///
+/// # Environment variables
+///
+/// Refer to the [documentation](https://learn.microsoft.com/en-us/sql/linux/sql-server-linux-configure-environment-variables)
+/// for a complite list of environment variables.
+///
+/// Following environment variables are required.
+/// A image provided by this module has default values for them.
+///
+/// ## `ACCEPT_EULA`
+///
+/// You need to accept the [End-User Licensing Agreement](https://go.microsoft.com/fwlink/?linkid=857698)
+/// before using the SQL Server image provided by this module.
+/// To accept EULA, you can set this environment variable to `Y`.
+/// The default value is `Y`.
+///
+/// ## `SA_PASSWORD`
+///
+/// The SA user password. This password is required to conform to the
+/// [strong password policy](https://learn.microsoft.com/en-us/sql/relational-databases/security/password-policy?view=sql-server-ver15#password-complexity).
+/// The default value is `yourStrong(!)Password`.
+///
+/// ## `MSSQL_PID`
+///
+/// The edition of SQL Server.
+/// The default value is `Developer`, which will run the container using the Developer Edition.
+#[derive(Debug)]
+pub struct MssqlServer {
+    env_vars: HashMap<String, String>,
+}
+
+impl MssqlServer {
+    const NAME: &'static str = "mcr.microsoft.com/mssql/server";
+    const TAG: &'static str = "2022-CU10-ubuntu-22.04";
+    const DEFAULT_SA_PASSWORD: &'static str = "yourStrong(!)Password";
```

Comment:
If we gonna use custom password as default, I think we should either expose this constant to allow reusing or add `with_password` method directly to `MsSqlServer` to make it easier to configure, otherwise it will be always necessary to use `RunnableImage` to override the password or hardcode from the sources.

Reply:
Thanks for the comment. Added `MssqlServer::with_password` method. In addition to that, changed `SA_PASSWORD` to `MSSQL_SA_PASSWORD` because `SA_PASSWORD` seems to be deprecated.

https://learn.microsoft.com/en-us/sql/linux/sql-server-linux-configure-environment-variables?view=sql-server-2017#environment-variables

> The SA_PASSWORD environment variable is deprecated. Use MSSQL_SA_PASSWORD instead.